### PR TITLE
feat: populate data rows and last upload in wireframe 2.1-2.2

### PR DESCRIPTION
**Pull request by signekb**
## Description

- This PR adds the feature of populating the fields data rows and last upload in wireframe 2.1-2.2.
- These changes are done since it has been a dummy value `Cell` until now.

image

- `data rows` has been populated by:
  - adding it as a field in the `Tables` model, setting it's default value to `0`,
  - then counting the rows in the file that is being uploaded and 
  - updating `data_rows` by adding those rows to the current `data_rows` count at data upload in the `project_id_metadata_id_data_update` view
  - (Counting the rows in the file was implemented in #171)
- `last upload` has been populated by:
  - adding it as a field in the `Tables`  model, setting it to `null` at creation
  - adding that it will only be updated on save when `data_rows` < 0 

In addition, I have made some small corrections:
- I changed the variable name tables to table since it only contains one Tables instance
- I split the test of modified_at being null at creation and not null after update into two tests
- I made some minor changes to a docstring
- I added the word "rows" to the successful upload message

## Related Issues

Closes #...

See also Issues #162 

## Testing

- [X] Yes

## Reviewer Focus

This PR requires an in-depth review.

## Checklist

For all PRs that are not general documentation

- [X] Tests accompany or reflect changes to the code
- [X] Tests ran and passed locally
- [X] Ran the linter and formatter
- [X] Build has passed locally

### CODE REVIEW

**Review by philter87 — DISMISSED** [review]
Yay! Looks good

File: sprout/views/projects_id_metadata_id_data_update.py [hunk 1]
```diff
@@ -51,11 +51,15 @@ def projects_id_metadata_id_data_update(
...
+        table.data_rows = table.data_rows + count_rows(new_server_file)
+        table.save()
+
         context = {
-            "table_name": tables.name,
+            "table_name": table.name,
             "upload_success": True,
             "file_metadata": files,
-            "number_rows": count_rows(new_server_file),
+            "number_rows": table.data_rows,
```
Comment (lwjohnst86) [inline sprout/views/projects_id_metadata_id_data_update.py:62]:
```suggestion
            "number_rows": new_rows_added,
```

this should be kept. It informs the user how many new rows have been added, not how many are currently there after the addition.

File: sprout/views/projects_id_metadata_id_data_update.py [hunk 2]
```diff
@@ -51,11 +51,15 @@ def projects_id_metadata_id_data_update(
         # )
         # TODO: verify that database has been written to.
 
+        # update tables model with new data rows
+        table.data_rows = table.data_rows + count_rows(new_server_file)
```
Comment (lwjohnst86) [inline sprout/views/projects_id_metadata_id_data_update.py:55]:
```suggestion
        new_rows_added = count_rows(new_server_file)
        table.data_rows = table.data_rows + new_rows_added
```

**Review by lwjohnst86 — DISMISSED** [review]
Some small edits ☺️

**Review by lwjohnst86 — APPROVED** [review]

File: sprout/views/projects_id_metadata_id_data_update.py (hunk 1 above)
Comment (signekb) [reply sprout/views/projects_id_metadata_id_data_update.py]: Ah, you're right! My bad

**Pull request merged**